engine/robot_state.py

```diff
@@ -0,0 +1,138 @@
+        radius: the radius of the robot
+    Preconditions:
+        x_pos is an int within the bounds of the user-defined traversal grid
+        y_pos is an int within the bounds of the user-defined traversal grid
+        heading is a float
+        epsilon is a float
+        max_velocity is a float
+        radius is a float
+    """
+    def __init__(self, x_pos, y_pos, heading, epsilon, max_velocity, radius):
+        """
+        Instance Attributes:
+            epsilon: dictates the turning radius of the robot. Lower epsilon results in tighter turning radius.
+            max_velocity: the maximum velocity of the robot
+            radius: the radius of the robot
+            width: width of the robot in cm
+            front_ultrasonic: the ultrasonic at the front of the robot, used for detecting obstacles
+            lf_ultrasonic: the ultrasonic at the front of the left side of the robot, used for boundary following
+            lb_ultrasonic: the ultrasonic at the back of the left side of the robot, used for boundary following
+            rf_ultrasonic: the ultrasonic at the front of the right side of the robot, used for boundary following
+            rb_ultrasonic: the ultrasonic at the back of the right side of the robot, used for boundary following
+            is_sim: False if the physical robot is being used, True otherwise
+            is_store: False if csv data should not be stored, True otherwise
+            position_kp: the proportional factor of the position PID
+            position_ki: the integral factor of the position PID
+            position_kd: the derivative factor of the position PID
+            position_noise: the flat amount of noise added to the robot's phase on each localization step
+            heading_kp: the proportional factor of the heading PID
+            heading_ki: the integral factor of the heading PID
+            heading_kd: the derivative factor of the heading PID
+            heading_noise: ?
+            phase: the phase of the robot
+            time_step: the amount of time that passes between each feedback loop cycle, should only be used if is_sim
+                is True
+            move_dist: the distance in meters that the robot moves per time dt
+            turn_angle: the angle in radians that the robot turns per time dt regardless of time step
+            plastic_weight: the weight of the trash the robot has collected
+            init_threshold (Double): Radius from initial position that will detect robot is back in initial position
+            goal_threshold (Double): Threshold from goal that will be detected as reaching goal in obstacle avoidance
+            noise_margin (Double): Margin from init_threshold that the robot has to leave before detecting robot has
+                left initial position
+        """
+        # User defined parameters
+        self.epsilon = epsilon
+        self.max_velocity = max_velocity
+        self.radius = radius
```

Comment:
Why do we need both radius and width? 
Also, the term radius is a little misleading because our robot is not a circle. Are we treating it as one?

Reply:
Just for context, all the attributes in `robot_state` are copied over from what currently exists in our initialization of `robot`.

I think radius and width was added by Allen for obstacle avoidance but I'm not too sure. @AC408 if u have any input?

Reply:
I added width for obstacle avoidance, not sure what radius is. Might be able to remove width and use radius

Reply:
ok radius must've been made back in the day. so quick w the response Allen LOL